support-frontend/assets/helpers/productCatalog.tsx

```diff
@@ -41,325 +45,278 @@ export function filterBenefitByRegion(
 	return true;
 }
 
-/**
- * TODO: make this stricter.
- * Currently we have this so that other places can use it,
- * but when we make it more constrained, we should pick up any type errors.
- */
-export type ProductId = string;
-
-export const productCatalogDescription: Record<string, ProductDescription> = {
-	/**
-	 * We need `SupporterPlusWithGuardianWeekly` for the the landing page while we migrate
-	 * away from the hacked TierThree product to the actual TierThree product below.
-	 */
-	SupporterPlusWithGuardianWeekly: {
-		label: 'Digital + print',
-		benefitsSummary: [
-			'The rewards from ',
-			{ strong: true, copy: 'All-access digital' },
-		],
-		benefits: [
-			{
-				copy: 'Guardian Weekly print magazine delivered to your door every week  ',
-				tooltip: `Guardian Weekly is a beautifully concise magazine featuring a handpicked selection of in-depth articles, global news, long reads, opinion and more. Delivered to you every week, wherever you are in the world.`,
-			},
-		],
-		deliverableTo: gwDeliverableCountries,
-		ratePlans: {
-			MonthlyWithGuardianWeekly: {
-				billingPeriod: 'Monthly',
-			},
-			AnnualWithGuardianWeekly: {
-				billingPeriod: 'Annual',
-			},
-			MonthlyWithGuardianWeeklyInt: {
-				billingPeriod: 'Monthly',
-			},
-			AnnualWithGuardianWeeklyInt: {
-				billingPeriod: 'Annual',
-			},
-		},
-	},
+export const productKeys = Object.keys(typeObject) as ProductKey[];
+export function isProductKey(val: unknown): val is ProductKey {
```

Comment:
candidate for [isOneOf](https://github.com/guardian/csnx/blob/32628d3d59087c58d39d8c0d3f6ee6b794e5a642/libs/%40guardian/libs/src/isOneOf/README.md)?

Reply:
Looking at [the code](https://github.com/guardian/csnx/blob/32628d3d59087c58d39d8c0d3f6ee6b794e5a642/libs/%40guardian/libs/src/isOneOf/isOneOf.ts), I do find it quite hard to read compared to the above. And probably way harder to change.

What benefit do you think we get from using it?

Reply:
In general I think it's worthwhile to use a function if it already exists, and type guards always make me think twice.

In this case I thought maybe it could get rid of an `as` or two that we're using here. But I see your point 👍